lib/msf/ui/console/command_dispatcher/modules.rb

```diff
@@ -633,11 +640,13 @@ def cmd_use(*args)
 
             begin
               mod = framework.modules.create(mod_name)
+
               unless mod
-                # Try one more time; see #4549
-                sleep CMD_USE_TIMEOUT
-                mod = framework.modules.create(mod_name)
-                unless mod
+                unless mod_resolved
+                  mods_found = cmd_search('-u', mod_name)
```

Comment:
Since this is effectively recursive, this implementation is potentially dangerous and could use a refactor and/or intensive testing.